docs/architecture/roomUtil.md

```diff
@@ -0,0 +1,36 @@
+# RoomUtil
+
+The [RoomUtil](../src/utils/roomUtil.ts) abstracts selecting the [RoomHandler](./roomHandler.md) for a given [GameState](../data/GameState.md),
+passing the [RoomState](../data/roomState.md) to the current RoomHandler,
+and finding a new GameState when the player leaves the current [Room](../data/Room.md).
+
+Since the RoomUtil has to handle the player leaving the room,
+it also has to aggregate  [CommandDescription](../data/commandDescription.md) along with the new GameState.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Anything
+  participant RU as RoomUtil
+  participant RH1 as RoomHandler<RoomId1>
+  participant RH2 as RoomHandler<RoomId2>
+
+  note over A,RU: type RoomState = Room1State | Room2State
+  A ->> RU: run(command: string, gameState: GameState)
```

Comment:
I don't think RoomUtil needs a run function. GameUtil can fetch and run the RoomHandler.

RoomUtil can still be responsible for coercing the new game state. GameUtil will call RoomUtil to do that. I'm trying to keep logic out of GameUtil, but in doing so I think I made RoomUtil overly complicated

Reply:
Updated